test/array.cpp

```diff
@@ -399,12 +399,16 @@ CAMP_TEST_BEGIN(array, structured_binding)
           a[1] == 3;
 } CAMP_TEST_END(array, structured_binding)
 
+#if defined(CAMP_ENABLE_CUDA) && (CAMP_CXX_STD == 17)
```

Comment:
This could probably be changed to `#if !defined(CAMP_ENABLE_CUDA)` with the `#else` removed. We already check if building with c++17 above: `#if defined(__cplusplus) && __cplusplus >= 201703L`